### PR TITLE
[WIP] Component Groups

### PR DESCRIPTION
**Pull request by jbrooksuk**
Opening this now so you can track the progress of it and contribute if wanted.

Screenshot:

screenshot 2015-01-01 14 34 58

All that's needed for a component group is a "name".

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
This will need rebasing at some point too.

File: app/database/migrations/2014_12_31_230032_CreateComponentGroupsTable.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+class CreateComponentGroupsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('component_groups', function (Blueprint $table) {
```
Comment (GrahamCampbell) [inline app/database/migrations/2014_12_31_230032_CreateComponentGroupsTable.php]: missing import for schema

File: app/models/Group.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+use Illuminate\Database\Eloquent\SoftDeletingTrait;
+use Watson\Validating\ValidatingTrait;
+
+class Group extends Eloquent
```
Comment (GrahamCampbell) [inline app/models/Group.php]: missing import for eloquent

File: app/database/migrations/2014_12_31_230032_CreateComponentGroupsTable.php (hunk 1 above)
Comment (jbrooksuk) [inline app/database/migrations/2014_12_31_230032_CreateComponentGroupsTable.php]: `use Illuminate\Database\Schema\Schema;`?

Comment (GrahamCampbell) [inline app/database/migrations/2014_12_31_230032_CreateComponentGroupsTable.php]: no

Comment (GrahamCampbell) [inline app/database/migrations/2014_12_31_230032_CreateComponentGroupsTable.php]: `use Illuminate\Support\Facades\Schema`

Comment (jbrooksuk) [inline app/database/migrations/2014_12_31_230032_CreateComponentGroupsTable.php]: Balls.

Comment (GrahamCampbell) [inline app/database/migrations/2014_12_31_230032_CreateComponentGroupsTable.php]: You've reference the actual instance there, but you want to call the "facade". It's basically a proxy to the container.

File: app/models/Group.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\SoftDeletingTrait;
+use Watson\Validating\ValidatingTrait;
+
+class Group extends Model
+{
+    use ValidatingTrait;
```
Comment (GrahamCampbell) [inline app/models/Group.php]: should be `use SoftDeletingTrait, ValidatingTrait`.

File: app/database/migrations/2014_12_31_230032_CreateComponentGroupsTable.php (hunk 1 above)
Comment (jbrooksuk) [inline app/database/migrations/2014_12_31_230032_CreateComponentGroupsTable.php]: Ah, gotcha.

File: src/Controllers/DashComponentController.php [hunk 4]
```diff
@@ -10,6 +10,30 @@
...
 {
+    protected $subMenu = [];
+    protected $subTitle = 'Components';
+
+    public function __construct()
+    {
+        $this->subMenu = [
+            'components' => [
+                'title'  => 'Components',
+                'url'    => URL::route('dashboard.components'),
```
Comment (GrahamCampbell) [inline src/Controllers/DashComponentController.php]: missing url import

File: app/models/Group.php (hunk 3 above)
Comment (jbrooksuk) [inline app/models/Group.php]: Done.

File: src/Controllers/DashComponentController.php (hunk 4 above)
Comment (jbrooksuk) [inline src/Controllers/DashComponentController.php]: Done.

Comment (GrahamCampbell) [inline src/Controllers/DashComponentController.php]: it's not alphabetically sorted. :(

**Comment by GrahamCampbell** [comment]
I can rebase this if you want?

**Comment by jbrooksuk** [comment]
Whoops, yeah, please. I messed it up.

**Comment by GrahamCampbell** [comment]
And the other one too?

**Comment by GrahamCampbell** [comment]
Done. Change quite a lot during the process too. For example, don't use `Lang::get` in views, use the `trans` function.

**Comment by jbrooksuk** [comment]
Sure :)

**Comment by GrahamCampbell** [comment]
Rebased.

**Comment by GrahamCampbell** [comment]
Someone has pushed hetre and deleded all my changes?

**Comment by jbrooksuk** [comment]
Pretty sure neither this nor team members was me.

**Comment by GrahamCampbell** [comment]
I'm working on this now...

**Comment by GrahamCampbell** [comment]
The group model seems to have a name, but the migration doesn't add that column?

File: app/views/dashboard/groups.blade.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
+@extends('layout.dashboard')
+
+@section('content')
+@if(isset($subMenu))
+@include('partials.dashboard.sub-sidebar')
+@endif
+<div class='content-panel'>
+	<div class="header">
+        <span class="uppercase">
```
Comment (GrahamCampbell) [inline app/views/dashboard/groups.blade.php]: dodgie indentation here

**Comment by jbrooksuk** [comment]
Ummm, one sec, lemme check it out.

**Comment by jbrooksuk** [comment]
I left this unfinished. Will continue work.

**Comment by jbrooksuk** [comment]
@GrahamCampbell I pulled down your changes, finished the work, then rebased the last three commits into one. Now I still need to force push?

**Comment by jbrooksuk** [comment]
I'm chuffed with that. It all seems to have worked!

**Comment by jbrooksuk** [comment]
They're implemented, but now we need a way of displaying the groups in the status page index. @joecohens I'll leave that with you, but I'll merge the functionality in.

File: app/views/dashboard/groups.blade.php (hunk 5 above)
Comment (GrahamCampbell) [inline app/views/dashboard/groups.blade.php]: still wrong

File: src/Models/Component.php [hunk 6]
```diff
@@ -43,7 +43,26 @@ class Component extends Model implements TransformableInterface
...
-    protected $fillable = ['name', 'description', 'status', 'user_id', 'tags', 'link', 'order'];
+    protected $fillable = [
+        'name',
+        'description',
+        'status',
+        'user_id',
+        'tags',
+        'link',
+        'order',
+        'group_id',
```
Comment (GrahamCampbell) [inline src/Models/Component.php]: you need to update the property annotations

File: src/Models/ComponentGroup.php [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+namespace CachetHQ\Cachet\Models;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\SoftDeletingTrait;
+use Watson\Validating\ValidatingTrait;
+
+/**
+ * @property int            $id
+ * @property int            $component_id
+ * @property int            $group_id
```
Comment (GrahamCampbell) [inline src/Models/ComponentGroup.php]: this property doesnt exist?

**Comment by GrahamCampbell** [comment]
Ping @jbrooksuk.

**Comment by jbrooksuk** [comment]
Will look now :)

**Comment by jbrooksuk** [comment]
Done.

**Comment by GrahamCampbell** [comment]
:beers:

**Pull request merged**